docs/scripts/build.js

```diff
@@ -0,0 +1,19 @@
+const { exec } = require('node:child_process');
+const args = process.argv.slice(2);
+
+const themesArgIndex = args.indexOf('--themes-dir');
+let buildCommand = 'exec-bin node_modules/.bin/hugo/hugo --gc --minify';
+
+if (themesArgIndex !== -1) buildCommand += ` --themesDir ${args[themesArgIndex + 1]}`;
+
+const buildProcess = exec(buildCommand, (error, stdout) => {
+    if (error) {
+        console.error(`exec error: ${error}`);
+        return;
+    }
+    console.log(`stdout: ${stdout}`);
+});
+
+buildProcess.stdout.on('data', function (data) {
+    console.log(data);
+});
```

Comment:
```suggestion
@@ -0,0 +1,19 @@
const { exec } = require('node:child_process');
const args = process.argv.slice(2);

const isDev = (args.indexOf('--dev') > 0);
const themesArgIndex = args.indexOf('--themes-dir');
let startServerCommand = isDev? 'exec-bin node_modules/.bin/hugo/hugo server' : 'exec-bin node_modules/.bin/hugo/hugo --gc --minify';;

if (themesArgIndex !== -1) startServerCommand += ` --themesDir ${args[themesArgIndex + 1]}`;

const startServer = exec(startServerCommand, (error, stdout) => {
    if (error) return console.error(`exec error: ${error}`);
    console.log(`stdout: ${stdout}`);
});

startServer.stdout.on('data', console.log);
```

Use one script for both build and server and delete server.js?

Reply:
Make sure to adjust the commands accordingly.